nucypher/acumen/perception.py

```diff
@@ -358,8 +378,65 @@ def record_remote_fleet_state(self,
                                   state_checksum: str,
                                   timestamp: maya.MayaDT,
                                   population: int):
+
+        if checksum_address not in self._current_state:
+            raise KeyError(f"A node {checksum_address} is not present in the current fleet state")
+
         nickname = Nickname.from_seed(state_checksum, length=1)
-        self.remote_states[checksum_address] = ArchivedFleetState(checksum=state_checksum,
-                                                                  nickname=nickname,
-                                                                  timestamp=timestamp,
-                                                                  population=population)
+        state = ArchivedFleetState(checksum=state_checksum,
+                                   nickname=nickname,
+                                   timestamp=timestamp,
+                                   population=population)
+
+        self._remote_last_seen[checksum_address] = maya.now()
+        self._remote_states[checksum_address] = state
+
+    def status_info(self, checksum_address_or_node: Union[ChecksumAddress, 'Ursula']) -> 'RemoteUrsulaStatus':
+
+        if isinstance(checksum_address_or_node, str):
+            node = self[checksum_address_or_node]
+        else:
+            node = checksum_address_or_node
+
+        recorded_fleet_state = self._remote_states.get(node.checksum_address, None)
+        last_learned_from = self._remote_last_seen.get(node.checksum_address, None)
+        worker_address = node.worker_address if node.verified_node else None
+
+        return RemoteUrsulaStatus(verified=node.verified_node,
+                                  nickname=node.nickname,
+                                  staker_address=node.checksum_address,
+                                  worker_address=worker_address,
+                                  rest_url=node.rest_url(),
+                                  timestamp=node.timestamp,
+                                  last_learned_from=last_learned_from,
+                                  recorded_fleet_state=recorded_fleet_state,
+                                  )
+
+
+class RemoteUrsulaStatus(NamedTuple):
+    verified: bool
+    nickname: Nickname
+    staker_address: ChecksumAddress
+    worker_address: Optional[ChecksumAddress]
+    rest_url: str
+    timestamp: maya.MayaDT
+    recorded_fleet_state: Optional[ArchivedFleetState]
+    last_learned_from: Optional[maya.MayaDT]
+
+    def to_json(self):
```

Comment:
```suggestion
    def to_json(self) -> Dict[str, str]:
```

Reply:
It's more of `Dict[str, Any]`. If we were being especially formal we could define a `JSON = Union[str, int, float, NoneType, list[JSON], dict[str, JSON]]` or something.